### PR TITLE
Buidlkite introduction: metricbeats

### PR DESCRIPTION
**Pull request by leo-ri**
## Proposed commit message

Introduction buildkite for metric beats

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Relates #1693

### CODE REVIEW

File: .buildkite/pull-requests.json [hunk 1]
```diff
@@ -15,6 +15,22 @@
...
+            "allowed_list": [ ],
+            "set_commit_status": true,
+            "build_on_commit": true,
+            "build_on_comment": true,
+            "trigger_comment_regex": "^/test metricbeat$",
+            "always_trigger_comment_regex": "^/test metricbeat$",
+            "skip_ci_labels": [  ],
+            "skip_target_branches": [ ],
+            "skip_ci_on_only_changed": [ ],
+            "always_require_ci_on_changed": [ ]
```
Comment (v1v) [inline .buildkite/pull-requests.json:33]: **question**: will this field be extended with the list of files that will always trigger the build?

```suggestion
            "always_require_ci_on_changed": [ "^filebeat/.*", "^x-pack/filebeat/.*", ]
```

[x-pack/filebeat](https://github.com/elastic/beats/blob/ead3090552292e39adb56e2a6e069b0787bbe83c/x-pack/filebeat/Jenkinsfile.yml#L3-L6) and [filebeat](https://github.com/elastic/beats/blob/main/filebeat/Jenkinsfile.yml#L3-L6) are the current list of files that will trigger a build. And those aliases can be found in [here](https://github.com/elastic/beats/blob/ead3090552292e39adb56e2a6e069b0787bbe83c/Jenkinsfile.yml#L21-L38)

File: .buildkite/pull-requests.json [hunk 2]
```diff
@@ -15,6 +15,22 @@
             "skip_target_branches": [ ],
             "skip_ci_on_only_changed": [ ],
             "always_require_ci_on_changed": [ ]
+        },
+        {
+            "enabled": true,
+            "pipelineSlug": "beats",
```
Comment (alexsapran) [inline .buildkite/pull-requests.json:21]: the slug needs to be changed to something like `betas-metricbeat`

File: .buildkite/pull-requests.json [hunk 3]
```diff
@@ -15,6 +15,22 @@
...
+            "allowed_list": [ ],
+            "set_commit_status": true,
+            "build_on_commit": true,
+            "build_on_comment": true,
+            "trigger_comment_regex": "^/test metricbeat$",
+            "always_trigger_comment_regex": "^/test metricbeat$",
+            "skip_ci_labels": [  ],
+            "skip_target_branches": [ ],
+            "skip_ci_on_only_changed": [ ],
+            "always_require_ci_on_changed": [ "^metricbeat/.*", "^Jenkinsfile", "^\\.ci/scripts/.*", "^go.mod", "^pytest.ini", "^dev-tools/.*", "^libbeat/.*", "^testing/.*"]
```
Comment (alexsapran) [inline .buildkite/pull-requests.json:33]: why do we need to trigger this for each Jenkinfile change?

**Review by jlind23 — APPROVED** [review]

**Comment by pierrehilbert** [comment]
Need to be backported to 8.11 as we are release 8.11.4

**Pull request merged**